tests/sources/test_sharepoint_online.py

```diff
@@ -548,10 +707,53 @@ async def test_site_drives(self, client, patch_scroll):
         assert returned_items == actual_items
 
     @pytest.mark.asyncio
-    async def test_drive_items_non_recursive(self, client, patch_fetch, patch_scroll):
-        """Basic setup for the test - no recursion through directories"""
-        pass
+    async def test_drive_items_delta(self, client, patch_fetch, patch_scroll_delta_url):
+        delta_url_input = "https://sharepoint.com/delta-link-lalal"
+        delta_url_next_sync = "https://sharepoint.com/delta-link-lalal/page-2"
+        items_page_1 = ["1", "2"]
+        items_page_2 = ["3", "4"]
 
+        patch_scroll_delta_url.return_value = AsyncIterator(
+            [
+                {"@odata.deltaLink": delta_url_next_sync, "value": items_page_1},
+                {"@odata.deltaLink": delta_url_next_sync, "value": items_page_2},
```

Comment:
this uses `delta_url_next_sync` for both items, but that variable should only be for page 2, right?

Reply:
Good one! Tbh I thought it's always returned - @vidok can you help us on that?

Reply:
Good catch! There should be `@odata.nextLink` and no `@odata.deltaLink` in the first item.